### PR TITLE
Fix hidden field validation

### PR DESCRIPTION
**Pull request by gnawf**
Tried to bump Nadel in Central Schema and I found an edge case. It's legal for all fields to be `@hidden` if the type itself is going to be deleted because all references to it are from `@hidden` fields.

### CODE REVIEW

File: lib/src/test/kotlin/graphql/nadel/validation/NadelFieldValidationTest.kt [hunk 1]
```diff
@@ -670,61 +671,90 @@ class NadelFieldValidationTest : DescribeSpec({
...
+                        }
+                    """.trimIndent(),
+                ),
+            )
+
+            val errors = validate(fixture)
+            assertTrue(errors.isEmpty())
+        }
+
+        it("passes if all fields are hidden and their container type is deleted") {
```
Comment (gnawf) [inline lib/src/test/kotlin/graphql/nadel/validation/NadelFieldValidationTest.kt:730]: New test here

File: lib/src/test/kotlin/graphql/nadel/validation/NadelFieldValidationTest.kt [hunk 2]
```diff
@@ -670,61 +671,90 @@ class NadelFieldValidationTest : DescribeSpec({
...
                 underlyingSchema = mapOf(
                     "test" to """
-                type Query {
-                    echo: String
-                }
-            """.trimIndent(),
+                        type Query {
+                            echo: String
+                        }
+                    """.trimIndent(),
```
Comment (gnawf) [inline lib/src/test/kotlin/graphql/nadel/validation/NadelFieldValidationTest.kt:695]: Reformat

File: lib/src/test/kotlin/graphql/nadel/validation/NadelFieldValidationTest.kt [hunk 3]
```diff
@@ -33,7 +34,7 @@ class NadelFieldValidationTest : DescribeSpec({
             )
 
             val errors = validate(fixture)
-            assert(errors.map { it.message }.isEmpty())
+            assertTrue(errors.map { it.message }.isEmpty())
```
Comment (gnawf) [inline lib/src/test/kotlin/graphql/nadel/validation/NadelFieldValidationTest.kt:37]: For some reason if you run individual tests with `kotlin.assert` it doesn't fail.

**Review by Juliano-Prado — APPROVED** [review]

**Review by bbakerman — APPROVED** [review]
